lib/_stream_readable.js

```diff
@@ -68,11 +68,19 @@ function ReadableState(options, stream) {
   // the point at which it stops calling _read() to fill the buffer
   // Note: 0 is a valid value, means "don't call _read preemptively ever"
   var hwm = options.highWaterMark;
-  var defaultHwm = this.objectMode ? 16 : 16 * 1024;
-  this.highWaterMark = (hwm || hwm === 0) ? hwm : defaultHwm;
+  if (hwm != null) {
```

Comment:
a zero watermark doesn't make sense, anything less than 1 is nothing getting processed.

also why delete `defaultHwm`?